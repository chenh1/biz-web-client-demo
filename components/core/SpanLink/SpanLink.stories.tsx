import type { SpanLinkType } from './SpanLink';
import React from 'react';
import { SpanLink } from './SpanLink';

type Args = {
  args: SpanLinkType
}

const Template = (args: SpanLinkType) => <SpanLink {...args}>SpanLink</SpanLink>;
export default {
  component: Template
}

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const StandardLink: Args = {
  args: {
    uncolored: false,
  },
};

export const UnstyledLink: Args = {
  args: {
    uncolored: true,
  },
};
