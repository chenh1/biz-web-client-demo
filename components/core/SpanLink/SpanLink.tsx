import type { GtmSet } from '../../constants/tagging';
import React, { useMemo } from 'react';
import { CLICK_CLASS, SECTION_VIEW_CLASS } from '../../constants/tagging';

const getClasses = (uncolored: boolean): `text-${string}` => {
  if (uncolored) {
    return "text-night-black dark:text-max-white"
  }

  return "text-apex-blue"
}

export type SpanLinkType = {
  uncolored?: boolean,
  children?: string | JSX.Element | JSX.Element[], 
} & GtmSet

export const SpanLink: React.FC<SpanLinkType> = ({ uncolored=false, gtmClick, gtmSectionView, dataGtmId, children }) => {
  const computedClasses = useMemo(() => {
    return getClasses(uncolored);
  }, [uncolored]);

  return (
    <span data-gtm-id={dataGtmId} className={`${gtmClick ? CLICK_CLASS : ''} ${gtmSectionView ? SECTION_VIEW_CLASS : ''} relative cursor-pointer font-body ${computedClasses} before:content-[""] before:absolute before:translate-y-[6px] before:bottom-0 before:duration-300 before:ease-in-out before:border-apex-blue before:w-0 before:border-2 before:opacity-0 hover:brightness-110 hover:before:w-full hover:before:opacity-100`}>
      {children}
    </span>
  )
};