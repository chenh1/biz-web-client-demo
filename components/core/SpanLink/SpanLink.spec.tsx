import { SpanLink } from "./SpanLink";
import { render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('renders a black span text link', async () => {
  const { asFragment } = render(
    <SpanLink uncolored={true}>Click here to go somewhere</SpanLink>
  );

  expect(asFragment()).toMatchSnapshot();
});


it('renders a traditional blue text link', async () => {
  const { asFragment } = render(
    <SpanLink>Click here to go somewhere</SpanLink>
  );

  expect(asFragment()).toMatchSnapshot();
});
