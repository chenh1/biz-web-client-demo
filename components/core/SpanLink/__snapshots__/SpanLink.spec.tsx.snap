// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a black span text link 1`] = `
<DocumentFragment>
  <span
    class="  relative cursor-pointer font-body text-night-black dark:text-max-white before:content-[""] before:absolute before:translate-y-[6px] before:bottom-0 before:duration-300 before:ease-in-out before:border-apex-blue before:w-0 before:border-2 before:opacity-0 hover:brightness-110 hover:before:w-full hover:before:opacity-100"
  >
    Click here to go somewhere
  </span>
</DocumentFragment>
`;

exports[`renders a traditional blue text link 1`] = `
<DocumentFragment>
  <span
    class="  relative cursor-pointer font-body text-apex-blue before:content-[""] before:absolute before:translate-y-[6px] before:bottom-0 before:duration-300 before:ease-in-out before:border-apex-blue before:w-0 before:border-2 before:opacity-0 hover:brightness-110 hover:before:w-full hover:before:opacity-100"
  >
    Click here to go somewhere
  </span>
</DocumentFragment>
`;
